projects/media-viewer/src/lib/icp/icp.service.ts

```diff
@@ -33,10 +27,11 @@ export class IcpDirective implements OnInit, OnDestroy {
               private readonly socketService: IcpUpdateService,
               private readonly presenterSubscriptions: IcpPresenterService,
               private readonly followerSubscriptions: IcpFollowerService,
-              private store: Store<IcpState>) {}
-
-  ngOnInit() {
-    this.subscription = this.toolbarEvents.icp.sessionLaunch.subscribe(() => this.launchSession());
+              private store: Store<IcpState>) {
+    this.store.pipe(select(fromIcpSelectors.getCaseId), filter(value => !!value)).subscribe(caseId => {
```

Comment:
this should probably also be added to the subscription, so that it can also be destroyed